File: spring-boot-starter-breuninger-jobs/src/main/java/com/breuninger/boot/jobs/domain/JobMarker.java
package com.breuninger.boot.jobs.domain;

import org.slf4j.Marker;
import org.slf4j.MarkerFactory;

public class JobMarker {

  public static final Marker JOB = MarkerFactory.getMarker("JOB");

  public static Marker jobMarker(final String jobType) {
    final var marker = MarkerFactory.getMarker(jobType);
    JOB.add(marker);
    return marker;
  }
}

File: jobs/src/main/kotlin/com/breuninger/boot/jobs/domain/JobMarker.kt
package com.breuninger.boot.jobs.domain

import org.slf4j.Marker
import org.slf4j.MarkerFactory

object JobMarker {

  private const val JOB = "JOB"

  val JOB_MARKER: Marker = MarkerFactory.getMarker(JOB)

  fun jobMarkerFor(jobId: JobId): Marker {
    val marker = MarkerFactory.getMarker(jobId.value)
    JOB_MARKER.add(marker)
    return marker
  }
}
